akka-remote-tests/src/multi-jvm/scala/akka/remote/RemoteFeaturesSpec.scala

```diff
@@ -0,0 +1,340 @@
+        val rar = remoteActorRef(remoteRole)
+        rar.isWatchIntercepted(watchee = rar, watcher = watcher) shouldBe false
+        rar.isWatchIntercepted(watchee = watchee, watcher = watcher) shouldBe false
+        enterBarrier("watch-not-intercepted")
+      }
+    }
+
+    "creation a remote actor from deployment config when remote features are disabled" in {
+      runOn(first) {
+        val sampleActor = system.actorOf(Props(new ProbeActor(probe.ref)), "sampleActor")
+        sampleActor.path.address shouldEqual node(second).address
+      }
+    }
+  }
+}
+
+abstract class RemotingFeaturesUnsafeSpec
+    extends RemotingFeaturesSpec(new RemotingFeaturesConfig(useUnsafe = true, artery = true)) {
+
+  import RemoteNodeDeathWatchSpec.Ack
+  import RemoteNodeDeathWatchSpec.DeathWatchIt
+  import RemoteNodeDeathWatchSpec.ProbeActor
+  import RemoteNodeDeathWatchSpec.UnwatchIt
+  import RemoteNodeDeathWatchSpec.WatchIt
+  import multiNodeConfig._
+
+  def stats(watcher: ActorRef, message: DeathWatchIt): Stats = {
+    probe.send(watcher, message)
+    probe.expectMsg(1.second, Ack)
+    provider.remoteWatcher.get ! Stats
+    expectMsgType[Stats]
+  }
+
+  "Remoting with UseUnsafeRemoteFeaturesWithoutCluster enabled" must {
+
+    "intercept and send system messages `Watch`/`Unwatch` to `RemoteWatcher` in the provider" in {
+      runOn(second) {
+        val watchee = system.actorOf(Props(new ProbeActor(probe.ref)), "watchee")
+        enterBarrier("watchee-started")
+        enterBarrier("watcher-started")
+        assertWatchIntercepted(identify(first, "watcher"), watchee, first)
+      }
+      runOn(first) {
+        enterBarrier("watchee-started")
+        val watcher = system.actorOf(Props(new ProbeActor(probe.ref)), "watcher")
+        enterBarrier("watcher-started")
+        assertWatchIntercepted(watcher, identify(second, "watchee"), second)
+      }
+      runOn(third, fourth) {
+        enterBarrier("watchee-started")
+        enterBarrier("watcher-started")
+        enterBarrier("watch-intercepted")
+        enterBarrier("watch-not-intercepted")
+      }
+      def assertWatchIntercepted(watcher: ActorRef, watchee: ActorRef, remoteRole: RoleName): Unit = {
+        val remoteWatcher = identifyWithPath(remoteRole, "system", "remote-watcher")
+        val rar = remoteActorRef(remoteRole)
+        rar.isWatchIntercepted(watchee = rar, watcher = watcher) shouldBe true
+        rar.isWatchIntercepted(watchee = rar, watcher = remoteWatcher) shouldBe true
+        enterBarrier("watch-intercepted")
+
+        rar.isWatchIntercepted(watchee = watchee, watcher = rar) shouldBe false
+        enterBarrier("watch-not-intercepted")
+      }
+    }
+
+    "`Watch` and `Unwatch` from `RemoteWatcher`" in {
+      runOn(first) {
+        val watcher = identify(first, "watcher")
+        val watchee = identify(second, "watchee")
+        awaitCond(stats(watcher, WatchIt(watchee)).watchingRefs == Set((watchee, watcher)), 2.seconds)
```

Comment:
I prefer `awaitAssert` over `awaitCond` because if it fails it gives better error reporting
```
awaitAssert {
  stats(watcher, WatchIt(watchee)).watchingRefs should === (Set((watchee, watcher)))
}
```